feature/experimental/system/gnmi/cliorigin/tests/mixed_oc_cli_origin_support_test/mixed_oc_cli_origin_support_test.go

```diff
@@ -0,0 +1,141 @@
+	t.Helper()
+	runningConfig, err := dut.RawAPIs().CLI(t).SendCommand(context.Background(), "show running-config")
+	if err != nil {
+		t.Fatalf("'show running-config' failed: %v", err)
+	}
+	return runningConfig
+}
+
+func testQoSWithCLIAndOCUpdates(t *testing.T, tCase testCase) {
+	dut := ondatra.DUT(t, "dut")
+
+	// We use ARISTA-specific command-line commands here. Other vendor devices won't accept it.
+	if dut.Vendor() != ondatra.ARISTA {
+		t.Skipf("Unsupported vendor device: got %v, expected %v", dut.Vendor(), ondatra.ARISTA)
+	}
+
+	qosPath := gnmi.OC().Qos()
+
+	t.Logf("Step 1: Make sure QoS queue under test is not already set.")
+	// Make sure the current config does not contain new data already.
+	if existingQueue := gnmi.LookupConfig(t, dut, qosPath.Queue(tCase.queueName).Config()); existingQueue.IsPresent() {
+		t.Fatalf("Detected an existing %v queue. This is unexpected.", tCase.queueName)
+	}
+
+	// Create OC addition to the config.
+	r := &oc.Root{}
+	qos := r.GetOrCreateQos()
+	qos.GetOrCreateQueue(tCase.queueName)
+	qos.GetOrCreateForwardingGroup(tCase.forwardGroupName).SetOutputQueue(tCase.queueName)
+
+	fptest.LogQuery(t, "QoS update for the OC config:", qosPath.Config(), qos)
+
+	t.Logf("Step 2: Retrieve current running-config")
+	runningConfig := showRunningConfig(t, dut)
+
+	newConfig, err := appendCLIConfig(dut, runningConfig, tCase.cliConfig)
+	if err != nil {
+		t.Fatalf("Error appending config to running-config: %v", err)
+	}
+
+	// Create and apply mixed CLI+OC SetRequest.
+	cliPath, err := schemaless.NewConfig[string]("", "cli")
+	if err != nil {
+		t.Fatalf("Failed to create CLI ygnmi query: %v", err)
+	}
+
+	t.Logf("Step 3: Send mixed-origin SetRequest")
+	mixedQuery := &gnmi.SetBatch{}
+	gnmi.BatchReplace(mixedQuery, cliPath, newConfig)
+	gnmi.BatchUpdate(mixedQuery, qosPath.Config(), qos)
+	result := mixedQuery.Set(t, dut)
+
+	t.Logf("gnmiClient.Set() response: %+v", result.RawResponse)
+
+	t.Logf("Step 4: Verify QoS queue configuration has been accepted by the target")
+	newRunningConfig := showRunningConfig(t, dut)
+	t.Logf("running config (-old, +new):\n%s", cmp.Diff(runningConfig, newRunningConfig))
+
+	// Validate
+	gotQueue := gnmi.GetConfig(t, dut, qosPath.Queue(tCase.queueName).Config())
+	if got := gotQueue.GetName(); got != tCase.queueName {
+		t.Errorf("Get(DUT queue name): got %v, want %v", got, tCase.queueName)
+	}
+	gotFG := gnmi.GetConfig(t, dut, qosPath.ForwardingGroup(tCase.forwardGroupName).Config())
+	if got := gotFG.GetName(); got != tCase.forwardGroupName {
+		t.Errorf("Get(DUT forwarding group name): got %v, want %v", got, tCase.forwardGroupName)
+	}
+	if got := gotFG.GetOutputQueue(); got != tCase.queueName {
+		t.Errorf("Get(DUT forwarding group output queue): got %v, want %v", got, tCase.queueName)
+	}
+}
+
+func TestQoSDependentCLI(t *testing.T) {
+	testQoSWithCLIAndOCUpdates(t, testCase{
+		cliConfig: `qos traffic-class 0 name target-group-BE0
+qos tx-queue 0 name BE0`,
```

Comment:
The value here is Arista EOS specific.  Other vendors may have 0 or more lines of cliConfig_test to be added to their `runningConfig` cli which will be added in followup PR

Reply:
I've added a vendor switch statement for handling different vendor CLIs in the re-opened PR: https://github.com/openconfig/featureprofiles/pull/1104